File: src/se/axelbengtsson/kinapokerscorecalculator/PlayType.java
package se.axelbengtsson.kinapokerscorecalculator;

import se.axelbengtsson.kinapokerscorecalculator.BonusType;
/**
 * Each round the player choose the playType
 */
public enum PlayType {
  Play(BonusType.NoBonus),
  NotPlay(BonusType.NotPlay),
  Automatic(BonusType.Automatic),
  Dragon(BonusType.Dragon),
  ColorDragon(BonusType.ColorDragon);

  final BonusType bonusType;
  PlayType(BonusType bonusType) {
    this.bonusType = bonusType;
  }
}

File: app/src/main/java/se/axel/bengtsson/kinapokerscorecalculator/PlayType.kt
package se.axel.bengtsson.kinapokerscorecalculator

/**
 * Each round the player choose the playType
 */
enum class PlayType(val bonusType: BonusType) {
  Play(BonusType.NoBonus),
  NotPlay(BonusType.NotPlay),
  Automatic(BonusType.Automatic),
  Dragon(BonusType.Dragon),
  ColorDragon(BonusType.ColorDragon);
}
